### PR TITLE
VL-122_Navigation-group-Typescript-migration_Vitalii-Dudnik

### PR DESCRIPTION
**Pull request by KinduD21**
https://ilevel.atlassian.net/jira/software/c/projects/VL/boards/11?selectedIssue=VL-122

### CODE REVIEW

File: src/ui.image-icon/types.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  name?: string;
+
+  /**
+   * Icon source (svg as a vue component).
+   */
+  src?: object;
+
+  /**
+   * Icon color.
+   * @values brand, grayscale, gray, red, orange, amber, yellow, lime, green, emerald, teal, cyan, sky, blue, indigo, violet, purple, fuchsia, pink, rose, black, white
```
Comment (itsJohnnyGrid) [inline src/ui.image-icon/types.ts:18]: remove @values

File: src/ui.image-icon/types.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    | "cyan"
+    | "sky"
+    | "blue"
+    | "indigo"
+    | "violet"
+    | "purple"
+    | "fuchsia"
+    | "pink"
+    | "rose"
+    | "white";
```
Comment (itsJohnnyGrid) [inline src/ui.image-icon/types.ts:41]: where is black?

File: src/ui.navigation-progress/UProgress.vue [hunk 3]
```diff
@@ -1,3 +1,62 @@
...
+import StepperProgress from "./StepperProgress.vue";
+import UHeader from "../ui.text-header/UHeader.vue";
+
+import type { UProgressProps } from "./types.ts";
+
+defineOptions({ inheritAttrs: false });
+
+const props = withDefaults(defineProps<UProgressProps>(), {
+  value: 0,
+  max: 100,
```
Comment (itsJohnnyGrid) [inline src/ui.navigation-progress/UProgress.vue:19]: max should be configurable

File: src/ui.navigation-progress/types.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+   */
+  dataTest?: string;
+}
+
+export interface StepperProgressProps {
+  progressPercent: string;
+
+  value: number;
+
+  color: string;
```
Comment (itsJohnnyGrid) [inline src/ui.navigation-progress/types.ts:73]: so add description also to this props

File: src/ui.navigation-progress/UProgress.vue [hunk 5]
```diff
@@ -16,7 +16,7 @@ defineOptions({ inheritAttrs: false });
 
 const props = withDefaults(defineProps<UProgressProps>(), {
   value: 0,
-  max: 100,
+  max: getDefault<UProgressProps>(defaultConfig, UProgress).max as number,
```
Comment (itsJohnnyGrid) [inline src/ui.navigation-progress/UProgress.vue:19]: for what reason `as number` is needed?

Comment (KinduD21) [reply src/ui.navigation-progress/UProgress.vue:19]: Discussed in private messages

**Review by itsJohnnyGrid — APPROVED** [review]

**Pull request merged**